gregBorchers/Homework2GregBorchers-probs1-4NoCards.js

```diff
@@ -0,0 +1,359 @@
+returns a string.  For example,  `fractionString(7,4)` should return "1 3/4",
+and `fractionString(3,3)` should (for now) return "1 0/3".  As before, assume
+that _n_ and _d_ are both positive integers.
+
+Note that _returning_ a string is not the same as _printing_ a string.  You may
+use `console.log()` for debugging, but your function should have the correct
+string as its return value. You can also print your return value with an
+expression like `console.log(fractionString(7,4))`. 
+*/
+
+var n =3;
+var d = 2;
+
+var fractionString = function(n,d) {
+  var wholeNumber = 0;
+  var numerator = 0;
+  var properFraction = "";
+
+  wholeNumber = Math.trunc( n / d );
+  numerator = n % d;
+
+  properFraction = wholeNumber + " " + numerator + "/" + d;
+
+  return properFraction;
+
+}
+
+var properFractionString = fractionString(n,d);
+console.log(properFractionString);
+
+/*
+**b)** Write a second version of your function which improves the output in two special cases:
+
+- A output string like "0 1/2" should be simplified to just "1/2";
+- A output string like "1 0/3" should be simplified to just "1".
+*/ 
+
+var fractionString2 = function(n,d) {
+  var wholeNumber = 0;
+  var numerator = 0;
+  var properFraction = "";
+
+  wholeNumber = Math.trunc( n / d );
+  numerator = n % d;
+
+  // general case format with non-zero values
+  properFraction = wholeNumber + " " + numerator + "/" + d;
+  
+  // handle special case formats for zero whole numbers
+  if (wholeNumber === 0) {
+    properFraction = numerator + "/" + d;
+  }
+  // handle special case formats for zero numerators
+  if (numerator === 0) {
+    properFraction = wholeNumber;
+  }
+
+  return properFraction;
+
+}
+
+// **************** Testing
+var n = 3;
+var d = 2;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+n = 0;
+d = 6;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+n = 4;
+d = 2;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+//divide by zero... 
+n = 3;
+d = 0;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+
+
+/*
+**2)** _(Moderate, 15%)_
+
+Rewrite each block below as the simplest equivalent you can discover.  Your
+equivalent should produce the same final conditions as the original code given
+the same initial conditions. In each case, if there is an undeclared variable,
+assume it has been declared earlier and set to an unknown value.
+
+//**2a)**
+*/
+
+var i;
+if ((n - Math.floor(n)) >= .5) {
+   i = Math.ceil(n);
+} else {
+  i = Math.floor(n);
+}
+
+// **************************  simplified form 2a)
+i = Math.trunc(n + .5);  // would be a simpler and shorter way to round n and store the rounded value in i
+
+
+//**2b)**
+
+var y, xIsFalse = (x? false : true);
+if (xIsFalse)
+   y = false;
+else
+   y = x;
+
+// **************************  simplified form 2b)
+var y = (x ? true : false);
```

Comment:
2b) Remember that in the equation given, y takes the value of x if x is true, rather than just becoming true if x is true.